src/libAtomVM/otp_socket.c

```diff
@@ -1570,8 +1624,7 @@ static term nif_socket_recv_with_peek(Context *ctx, struct SocketResource *rsrc_
     GlobalContext *global = ctx->global;
 
     int flags = MSG_WAITALL;
-    // TODO parameterize buffer size
-    ssize_t res = recvfrom(rsrc_obj->fd, NULL, DEFAULT_BUFFER_SIZE, MSG_PEEK | flags, NULL, NULL);
+    ssize_t res = recvfrom(rsrc_obj->fd, NULL, rsrc_obj->buf_size, MSG_PEEK | flags, NULL, NULL);
```

Comment:
We should use len, and I'm sorry I missed it when implementing len below.
What we could do is if len is 0, run getsockopt to get the current buffer size and use that.